File: fizzbuzz.js
class FizzBuzz {
    static calc(n) {
        if (n % 5 == 0 && n % 3 == 0)
            return "FizzBuzz";
        if (n % 3 == 0)
            return "Fizz";
        if (n % 5 == 0)
            return "Buzz";
        return n + '';
    }
}

$(function() {
    for (var i = 0; i < 30; i++) {
        var val = FizzBuzz.calc(i + 1);
        console.log($(`<p>${val}</p>`));
        $('#result').append($(`<p>${val}</p>`));
    }
});

File: fizzbuzz.exs
defmodule FizzBuzz do

defp calc x do
  case {rem(x, 3),rem(x, 5)} do
    {0,0} -> "FizzBuzz"
    {0,_} -> "Fizz"
    {_,0} -> "Buzz"
    _ -> "#{x}"
  end
end

defp print(n, limit) when n >= limit do
  IO.puts calc(n)
end

defp print n, limit do
  IO.puts calc(n)
  print(n + 1, limit)
end

def run limit do
  print 1, limit
end
end

FizzBuzz.run(30)
